homeassistant/components/sia/alarm_control_panel.py

```diff
@@ -0,0 +1,288 @@
+    """Class for SIA Alarm Control Panels."""
+
+    def __init__(
+        self,
+        entity_id: str,
+        name: str,
+        port: int,
+        account: str,
+        zone: int,
+        ping_interval: int,
+    ):
+        """Create SIAAlarmControlPanel object."""
+        self.entity_id = ALARM_FORMAT.format(entity_id)
+        self._unique_id = entity_id
+        self._name = name
+        self._port = port
+        self._account = account
+        self._zone = zone
+        self._ping_interval = get_ping_interval(ping_interval)
+        self._event_listener_str = f"{SIA_EVENT}_{port}_{account}"
+        self._unsub = None
+
+        self._should_poll = False
+        self._is_available = True
+        self._remove_unavailability_tracker = None
+        self._state = None
+        self._old_state = None
+        self._attr = {
+            CONF_ACCOUNT: self._account,
+            CONF_PING_INTERVAL: self.ping_interval,
+            CONF_ZONE: self._zone,
+            ATTR_LAST_MESSAGE: None,
+            ATTR_LAST_CODE: None,
+            ATTR_LAST_TIMESTAMP: None,
+        }
+
+    async def async_added_to_hass(self):
+        """Once the panel is added, see if it was there before and pull in that state."""
+        await super().async_added_to_hass()
+        state = await self.async_get_last_state()
+        _LOGGER.debug(
+            "Loading last state: %s",
+            state.state if state is not None and state.state is not None else "None",
+        )
+        if (
+            state is not None
+            and state.state is not None
+            and state.state
+            in [
+                STATE_ALARM_ARMED_AWAY,
+                STATE_ALARM_ARMED_CUSTOM_BYPASS,
+                STATE_ALARM_ARMED_NIGHT,
+                STATE_ALARM_DISARMED,
+                STATE_ALARM_TRIGGERED,
+                STATE_UNKNOWN,
+            ]
+        ):
+            self.state = state.state
+        else:
+            self.state = None
+        await self._async_track_unavailable()
+        async_dispatcher_connect(
+            self.hass, DATA_UPDATED, self._schedule_immediate_update
+        )
+        self._unsub = self.hass.bus.async_listen(
+            self._event_listener_str, self.async_handle_event
+        )
+        self.async_on_remove(self._sia_on_remove)
+
+    @callback
+    def _sia_on_remove(self):
+        """Remove the unavailability and event listener."""
+        if self._unsub:
+            self._unsub()
+        if self._remove_unavailability_tracker:
+            self._remove_unavailability_tracker()
+
+    async def async_handle_event(self, event: Event):
+        """Listen to events for this port and account and update states.
+
+        If the port and account combo receives any message it means it is online and can therefore be set to available.
+        """
+        await self.assume_available()
+        if int(event.data[EVENT_ZONE]) == self._zone:
```

Comment:
```suggestion
        if int(event.data[EVENT_ZONE]) != self._zone:
            return
```

and then outdent the block below